task4.py

```diff
@@ -0,0 +1,34 @@
+# to find n-th element of fibonacci series 
+num = int(input("Enter the number to see that which number is that position : "))
+def func_for_fib(num):
+    if num == 0 or num == 1:
+        return num
+    return func_for_fib(num-1) + func_for_fib(num-2)
+print(func_for_fib(num))
+
+
+# solution of reducing time complexity
+dictinary_for_store_fib_value = { 0 : 0 , 1 : 1}
+
+num1 = int(input("Enter the number to find the number at that position :"))
+def func_for_fib_sol(num):
+    if num in dictinary_for_store_fib_value:
+        return dictinary_for_store_fib_value[num]
+    
+    dictinary_for_store_fib_value[num] = func_for_fib_sol(num-1) + func_for_fib_sol(num-2)
```

Comment:
This will not reduce the time complexity. Try for n=30, n=60, and n=90, It will give no output to you. Please first try by yourself and still if you don't find any solution then it is fine, but don't copy paste the code. It's a humble request.